### PR TITLE
grab stateToken from window.oktaData for Okta-hosted custom SIW

### PR DESCRIPTION
**Pull request by sherizada-okta**
## Description:
use stateToken from window.oktaData if available (only set for custom Okta hosted SIW) to prevent customers who override config options and lose the stateToken set in config

prevents this scenario
1. Edit code in Sign in page code editor 
2. Set `config` to a new object without carrying over stateToken ` config = { ...myNewOptions } `
3. Sign into some app with a custom domain set
  a. on OIE, user will not be able to log in
  b. on Classic, user will not be redirected to correct app

## PR Checklist

- [ ] Have you verified the basic functionality for this change?
- [ ] Did you add tests, as appropriate, following our [Automated Test guidelines](https://oktawiki.atlassian.net/wiki/spaces/eng/pages/2676497890/Automated+Testing+in+the+Signin+Widget)?
- [ ] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?
- [ ] Did you verify the change by running [downstream monolith artifact](https://oktawiki.atlassian.net/wiki/spaces/eng/pages/102897979/Sign-in+Widget+Development#Sign-inWidgetDevelopment-Instructionstocreateandrunthedownstreamartifact(d16t))? (Provide link to build below)
- [ ] Does this PR include noticeable changes to the UI? (If yes, attach screenshots/video below)

### Issue:

- [OKTA-561119](https://oktainc.atlassian.net/browse/OKTA-561119)

### CODE REVIEW

File: src/widget/OktaSignIn.ts [hunk 1]
```diff
@@ -42,6 +42,10 @@ export default class OktaSignIn implements OktaSignInAPI {
       `);
 
     this.options = options;
+    if (window?.oktaData?.signIn?.stateToken) {
+      // Okta hosted custom SIW
+      this.options.stateToken = window?.oktaData?.signIn?.stateToken;
```
Comment (sherizada-okta) [inline src/widget/OktaSignIn.ts:47]: I am also contemplating setting the stateToken in the Admin sign in page code editor:
```
config.stateToken = oktaData.signIn.stateToken; // required
```

File: src/widget/OktaSignIn.ts [hunk 2]
```diff
@@ -42,6 +48,10 @@ export default class OktaSignIn implements OktaSignInAPI {
...
```
Comment (lesterchoi-okta) [inline src/widget/OktaSignIn.ts:53]: add check that `stateToken` isn't already set

Comment (lesterchoi-okta) [reply src/widget/OktaSignIn.ts:53]: nit: `window.oktaData.signIn.stateToken`

**Review by lesterchoi-okta — CHANGES_REQUESTED** [review]

File: src/models/Settings.ts [hunk 3]
```diff
@@ -26,14 +26,22 @@ import Logger from 'util/Logger';
...
 import CountryUtil from 'util/CountryUtil';
 import { OktaAuth } from '@okta/okta-auth-js';
+import { OktaData } from '../types';
+
 const SharedUtil = internal.util.Util;
 const assetBaseUrlTpl = hbs('https://global.oktacdn.com/okta-signin-widget/{{version}}');
 
+declare global {
+  interface Window {
+    oktaData: OktaData;
```
[the remark this answers is not in the document]
Comment (sherizada-okta) [reply src/models/Settings.ts]: yes, done.

**Review by lesterchoi-okta — APPROVED** [review]
:shipit:

**Pull request merged**